src/main/java/Commands/Command.java

```diff
@@ -12,17 +13,19 @@
     protected final User initiator; // who
     protected final ClientInterface client; // where
     protected final String[] args; // what
+    protected final DataAccessInterface api; // How
 
     /**
      * Basic constructor for a Command.
      * @param initiator is "who" called this Command
      * @param client is "where" the call came from
      * @param args is "what" to do
      */
-    public Command(User initiator, ClientInterface client, String[] args) {
+    public Command(User initiator, ClientInterface client, String[] args, DataAccessInterface api) {
```

Comment:
Consider changing the order of the <api> parameter. It feels more natural to have <args> as the last parameter.